tracing.go

```diff
@@ -626,3 +634,27 @@ func spanFromContext(ctx context.Context) *Span {
 	}
 	return nil
 }
+
+// ErrTransactionAlreadyInProgress is returne when we try to start a transaction
+// when another one is in progress.
+var ErrTransactionAlreadyInProgress = errors.New("transaction already in progress")
+
+// StartTransaction will create a transaction (root span) if there's no existing
+// transaction in the context.
+func StartTransaction(ctx context.Context, name string, options ...SpanOption) *Span {
+	currentTransaction := ctx.Value(spanContextKey{})
+	if currentTransaction != nil {
+		panic(ErrTransactionAlreadyInProgress)
```

Comment:
Not sure if I like the idea of panicing here, we shouldn't cause possible runtime issues with our SDK methods.

Could we return `err` instead here? And have users handle this?

Reply:
Right, I can return an error instead here. I'll be back to asking if I can `panic` in the HTTP handler though, and so, maybe you can shed some light on how other SDKs are dealing with this if that's a case that can happen?

Could we silently return the existing transaction with a `bool` to say it already was in progress instead? Could we just return `nil` and the user would have test for the transaction or get the existing one themselves?

Reply:
Yeah so there are the two options

1. We return `err`, so the users have to grab the existing transaction themselves.
2. We return the active transaction and don't create one. This does mean that the transaction won't have the context supplied in the constructor.

Option 2 is more "magic" behavior, but I think it's correct since two transactions running at once is basically useless to a person, so let us maybe do that. 

> Could we silently return the existing transaction with a bool to say it already was in progress instead

Sounds good to me, I think that's cleaner API